client/signup/steps/domains/index.jsx

```diff
@@ -120,6 +120,11 @@ const DomainsStep = React.createClass( {
 					domain: suggestion.domain_name,
 					productSlug: suggestion.product_slug
 				} )
+				: undefined,
+			privacyItem = isPurchasingItem && this.props.domainsWithPlansOnly
```

Comment:
I think this needs to be broadened a bit - what about the case when user is not affected by DWPO (an "old" user), but buys a custom domain and selects a Personal (or higher) plan. We should also add the privacy protection for them by default. In other words, when the user is buying a domain from their domain credit, we should check privacy protection for them (regardless of DWPO).